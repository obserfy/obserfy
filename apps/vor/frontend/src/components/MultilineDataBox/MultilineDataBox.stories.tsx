import { action } from "@storybook/addon-actions"
import { Meta, Story } from "@storybook/react"
import MultilineDataBox, { MultilineDataBoxProps } from "./MultilineDataBox"

export default {
  title: "Core/MultilineDataBox",
  component: MultilineDataBox,
  parameters: {
    componentSubtitle: "Just a simple MultilineDataBox",
  },
} as Meta

const Template: Story<MultilineDataBoxProps> = (args) => (
  <MultilineDataBox {...args} />
)

export const Default = Template.bind({})
Default.args = {
  label: "",
  value: "",
  onEditClick: action("edit clicked"),
  placeholder: "",
}
