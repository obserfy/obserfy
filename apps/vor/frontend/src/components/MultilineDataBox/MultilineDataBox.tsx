import { Trans } from "@lingui/macro"
import { useLingui } from "@lingui/react"
import { FC } from "react"
import { Box, Button, Flex } from "theme-ui"
import { borderBottom } from "../../border"
import { ReactComponent as EditIcon } from "../../icons/edit.svg"
import Icon from "../Icon/Icon"
import Markdown from "../Markdown/Markdown"
import { Typography } from "../Typography/Typography"

export interface MultilineDataBoxProps {
  label: string
  value: string
  onEditClick?: () => void
  placeholder: string
}
export const MultilineDataBox: FC<MultilineDataBoxProps> = ({
  label,
  value,
  onEditClick,
  placeholder,
}) => {
  const { i18n } = useLingui()

  return (
    <Box sx={{ alignItems: "flex-start" }}>
      <Flex sx={{ ...borderBottom, alignItems: "center" }}>
        <Typography.Body m={3} sx={{ fontWeight: "bold" }}>
          <Trans id={label} />
        </Typography.Body>
        <Button
          variant="text"
          ml="auto"
          mr={3}
          p={2}
          onClick={onEditClick}
          aria-label={`edit-${label.toLowerCase()}`}
          data-cy={`edit-${label.toLowerCase()}`}
          sx={{ color: "textMediumEmphasis" }}
        >
          <Icon as={EditIcon} />
        </Button>
      </Flex>

      <Markdown markdown={value || i18n._(placeholder)} mx={3} mt={3} mb={2} />
    </Box>
  )
}

export default MultilineDataBox
