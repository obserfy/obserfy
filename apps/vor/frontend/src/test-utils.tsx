/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
