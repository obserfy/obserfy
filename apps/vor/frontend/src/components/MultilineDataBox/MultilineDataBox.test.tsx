import { render } from "../../test-utils"
import { Default } from "./MultilineDataBox.stories"

describe("MultilineDataBox", () => {
  it("should render correctly", () => {
    const { container } = render(<Default {...Default.args} />)
    expect(container).toMatchSnapshot()
  })
})
