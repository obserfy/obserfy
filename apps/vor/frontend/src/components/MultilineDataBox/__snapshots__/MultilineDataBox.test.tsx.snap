// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultilineDataBox should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom-width: 1px;
  border-bottom-style: solid;
  border-bottom-color: var(--theme-ui-colors-border);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin: 16px;
  font-weight: bold;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  margin-left: auto;
  margin-right: 16px;
  padding: 8px;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 16px;
  margin-bottom: 8px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <p
          class="emotion-3"
        />
        <button
          aria-label="edit-"
          class="emotion-4"
          data-cy="edit-"
        >
          <div
            class="emotion-5"
          />
        </button>
      </div>
      <div
        class="prose prose-sm max-w-none undefined emotion-6"
      >
        <p />
      </div>
    </div>
  </div>
</div>
`;
